src/Functions/FunctionBase58Conversion.h

```diff
@@ -0,0 +1,232 @@
+        }
+
+        dst_data.resize(dst_pos - dst);
+    }
+};
+
+struct Base58Decode
+{
+    static constexpr auto name = "base58Decode";
+
+    static void process(const ColumnString& input, ColumnString::MutablePtr& dst_column, std::string& alphabet, size_t input_rows_count)
+    {
+        auto & dst_data = dst_column->getChars();
+        auto & dst_offsets = dst_column->getOffsets();
+
+        size_t current_allocated_size = input.getChars().size();
+
+        dst_data.resize(current_allocated_size);
+        dst_offsets.resize(input_rows_count);
+
+        const ColumnString::Offsets & src_offsets = input.getOffsets();
+
+        const auto * source = input.getChars().raw_data();
+        auto * dst = dst_data.data();
+        auto * dst_pos = dst;
+
+        size_t src_offset_prev = 0;
+        size_t processed_size = 0;
+
+        const auto& decoder = (alphabet == "bitcoin") ? Base58::bitcoin() :
+                             ((alphabet == "flickr") ? Base58::flickr() :
+                             ((alphabet == "ripple") ? Base58::ripple() :
+                                                       Base58::base58()));
+
+        std::string decoded;
+        for (size_t row = 0; row < input_rows_count; ++row)
+        {
+            size_t srclen = src_offsets[row] - src_offset_prev - 1;
+            /// Why we didn't use char* here?
+            /// We don't know the size of the result string beforehand (it's not byte-to-byte encoding),
+            /// so we may need to do many resizes (the worst case -- we'll do it for each row)
+            /// This way we do exponential resizes and one final resize after whole operation is complete
+            decoded.clear();
+            if (srclen)
+                decoder.decode(decoded, source, srclen);
+            size_t outlen = decoded.size();
+
+            if (processed_size + outlen >= current_allocated_size)
+            {
+                current_allocated_size += current_allocated_size;
+                dst_data.resize(current_allocated_size);
+                auto processed_offset = dst_pos - dst;
+                dst = dst_data.data();
+                dst_pos = dst;
+                dst_pos += processed_offset;
+            }
+            std::memcpy(dst_pos, decoded.c_str(), ++outlen);
+
+            source += srclen + 1;
+            dst_pos += outlen;
+
+            dst_offsets[row] = dst_pos - dst;
+            src_offset_prev = src_offsets[row];
+            processed_size += outlen;
+        }
+
+        dst_data.resize(dst_pos - dst);
+    }
+};
+
+template <typename Func>
+class FunctionBase58Conversion : public IFunction
+{
+public:
+    static constexpr auto name = Func::name;
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionBase58Conversion>();
+    }
+
+    String getName() const override
+    {
+        return Func::name;
+    }
+
+    bool isVariadic() const override { return true; }
```

Comment:
Why is base58*() variadic? Just asking because base64*() isn't and I don't really see a natural semantics in something like `base58Decode(s1, s2, s3)`

EDIT: Ah, the second parameter is the alphabet ... okay than.